src/components/layout/editor-components/common-properties-editor.tsx

```diff
@@ -0,0 +1,283 @@
+      </>
+    );
+  }
+
+  /**
+   * Render layout background color editor
+   */
+  private renderLayoutBackgroundColor = () => {
+    const { classes } = this.props;
+    return (
+      <div className={ classes.backgroundPickerContainer }>
+        <Typography variant="h4">{ strings.layoutEditor.commonComponents.backgroundColor }</Typography>
+        <ColorPicker
+          property={ this.getProperty(LayoutPropKeys.LayoutBackgroundColor, PageLayoutViewPropertyType.Color) }
+          onColorChange={ this.onSingleValueChange }
+        />
+        <Divider variant="fullWidth" color="rgba(0,0,0,0.1)" style={{ marginTop: theme.spacing(2), marginBottom: theme.spacing(2) }} />
+      </div>
+    );
+  }
+
+  /**
+   * Render layout padding editor
+   */
+  private renderLayoutPadding = () => {
+    return (
+      <div style={{ padding: theme.spacing(1) }}>
+        <Typography style={{ marginBottom: theme.spacing(1) }} variant="h4">{ strings.layoutEditor.commonComponents.paddings.title }</Typography>
+        <MarginPaddingEditor
+          itemKey="layout_padding"
+          properties={ this.getPaddingOrMarginProperties(LayoutPaddingPropKeys) }
+          onSingleValueChange={ this.onSingleValueChange }
+          onMultipleValueChange={ this.onMultipleValueChange }
+        />
+      </div>
+    );
+  }
+
+  /**
+   * Render layout margin editor
+   */
+  private renderLayoutMargin = () => {
+    return (
+      <div style={{ padding: theme.spacing(1) }}>
+        <Typography style={{ marginBottom: theme.spacing(1) }} variant="h4">{ strings.layoutEditor.commonComponents.margins.title }</Typography>
+        <MarginPaddingEditor
+          itemKey="layout_margin"
+          properties={ this.getPaddingOrMarginProperties(LayoutMarginPropKeys) }
+          onSingleValueChange={ this.onSingleValueChange }
+          onMultipleValueChange={ this.onMultipleValueChange }
+        />
+      </div>
+    );
+  }
+
+  /**
+   * Render layout gravity editor
+   */
+  private renderLayoutGravity = () => {
+    return (
+      <div style={{ padding: theme.spacing(1) }}>
+        <Typography variant="h4">{ strings.layoutEditor.commonComponents.layoutGravity }</Typography>
+        <GravityEditor
+          property={ this.getProperty(LayoutPropKeys.LayoutGravity, PageLayoutViewPropertyType.String) }
+          onSingleValueChange={ this.onSingleValueChange }
+        />
+        <Divider variant="fullWidth" color="rgba(0,0,0,0.1)" style={{ marginTop: theme.spacing(2), marginBottom: theme.spacing(2) }} />
+      </div>
+    );
+  }
+
+  /**
+   * Generic handler for single page layout property value changes
+   */
+  private onSingleValueChange = (updatedPageLayoutView: PageLayoutViewProperty) => {
+    const { onLayoutViewChange, pageLayoutView } = this.props;
+    const layoutViewToUpdate = JSON.parse(JSON.stringify(pageLayoutView)) as PageLayoutView;
```

Comment:
I suppose this is the solution for props property unexpectedly updating when extracted variable updated?
If so, ignore this, otherwise unnecessary JSON parsing.